### PR TITLE
Batch sync and delete resources

### PR DESCRIPTION
**Pull request by lulf**
Add a syncer and deleter goroutine for each infra, which is responsible
for syncing and deleting entities in batches. This speeds up the
create/delete performance test by 2x on my laptop.

### CODE REVIEW

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated-shared-infra

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated-shared-infra crc

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated-shared-infra

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated-shared-infra crc

File: pkg/state/errors.go [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
-}
-
-func NewNotInitializedError() error {
-	return &NotInitializedError{}
-}
+var (
+	NotInitializedError error = fmt.Errorf("not initialized")
+	NotSyncedError      error = fmt.Errorf("not synchronized")
+	NoEndpointsError    error = fmt.Errorf("no endpoints")
+	NotDeletedError     error = fmt.Errorf("error deleting")
```
Comment (k-wall) [inline pkg/state/errors.go:15]: `NotDeletedError` name seems to disagree the text.  DeletionError?

File: pkg/state/infra.go [hunk 2]
```diff
@@ -432,11 +536,159 @@ func (i *infraClient) freePortsInternal(endpoint *v1beta2.MessagingEndpoint) {
...
+			endpoint := req.resource.(*v1beta2.MessagingEndpoint)
+			result, err := i.buildRouterEndpointEntities(endpoint)
+			if err != nil {
+				req.done <- err
+				continue
+			}
+			routerEntities = append(routerEntities, result...)
+			built = append(built, req)
+		default:
+			log.Printf("unexpected value with type %T", v)
```
Comment (k-wall) [inline pkg/state/infra.go:675]: send an error to req.done?

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated-shared-infra

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated-shared-infra

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated-shared-infra

**Comment by lulf** [comment]
Hmm, another race here. If router is restarted and command client is reconnected before infra client got a connection error, we end up in a bad state. I think maybe the simplest fix for now is to reconnect on any error to ensure we have a synced state.

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated-shared-infra

**Comment by lulf** [comment]
@enmasse-ci run tests profile=isolated-shared-infra

**Pull request merged**